lib/quicly.c

```diff
@@ -3386,7 +3464,8 @@ static int do_send(quicly_conn_t *conn, quicly_send_context_t *s)
     if (ret == QUICLY_ERROR_SENDBUF_FULL)
         ret = 0;
     if (ret == 0) {
-        conn->egress.send_ack_at = INT64_MAX; /* we have sent ACKs for every epoch (or before address validation) */
+        if (conn->application == NULL || conn->application->super.unacked_count == 0)
+            conn->egress.send_ack_at = INT64_MAX; /* we have sent ACKs for every epoch (or before address validation) */
```

Comment:
I need to think a bit more if this change is correct.

Since f9e4400, we are seeing frequent assertion failures in CI under severe packet loss. Pasted below is a backtrace of that assertion. It could be the case that this error is due to us failing to reset `send_ack_at`.
```
[4657] Assertion failed: (now < conn->egress.loss.alarm_at), function assert_consistency, file /Users/travis/build/h2o/quicly/lib/quicly.c, line 580.
28542[4657] received fatal signal 6
28543[4657] 0   cli                                 0x0000000108e1a744 on_sigfatal + 100
28544[4657] 1   libsystem_platform.dylib            0x00007fffc7259b3a _sigtramp + 26
28545[4657] 2   ???                                 0x0000000000000000 0x0 + 0
28546[4657] 3   libsystem_c.dylib                   0x00007fffc70de420 abort + 129
28547[4657] 4   libsystem_c.dylib                   0x00007fffc70a5893 basename_r + 0
28548[4657] 5   cli                                 0x0000000108e0cdbe assert_consistency + 174
28549[4657] 6   cli                                 0x0000000108e0aff7 quicly_send + 1511
28550[4657] 7   cli                                 0x0000000108e1b731 send_pending + 81
28551[4657] 8   cli                                 0x0000000108e1a100 main + 5376
28552[4657] 9   libdyld.dylib                       0x00007fffc704a235 start + 1
```

Note that we would see much less CI failures as of 3e4267f, the commit that incorporates the fix for #248.

Reply:
Sadly not. Still working on it.

Reply:
This bug would be resolved separately in #253.